server/decode.go

```diff
@@ -85,6 +96,32 @@ func ProcessProtobufArchive(rootType string, archive []byte) (*desc.MessageDescr
 	return rootMD, nil
 }
 
+// truncateProtoDirectories attempts to locate a .proto file in the shortest path of a directory tree so that
+// import paths work correctly
+func truncateProtoDirectories(files map[string]string, rootDir string) map[string]string {
+	cleaned := make(map[string]string)
+
+	var zipBaseDir string
+
+	for filePath, contents := range files {
+		// Only need to do this once
+		if zipBaseDir == "" {
+			// Strip out zip base directory, which will look like "batchcorp-schemas-9789dfg70s980fdsfs"
+			parts := strings.Split(filePath, "/")
+			zipBaseDir = parts[0] + "/"
+		}
+
+		if !strings.Contains(filePath, zipBaseDir+rootDir+"/") {
```

Comment:
Just for my own understanding - this is the line that will cause a local .proto that imports another .proto from the same dir to not work, right?

Reply:
This line is to exclude protos outside of the root proto path that the user specifies. Like in schemas where we have the samples directory